CMake error when an already configured wearables is re-configured

**Describe the bug**

If I run `cmake .` again in an already created built environment, the configuration fails with:
~~~
-- Found YARP: /home/gitpod/mambaforge/envs/wearablesdev/lib/cmake/YARP (found version "3.7.2~dev")
--  [x] Plugin: iwear_logger (ENABLE_iwear_logger)
CMake Error at modules/IWearFrameVisualizer/CMakeLists.txt:22 (find_package):
  By not providing "FindIWear.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "IWear", but
  CMake did not find one.

  Could not find a package configuration file provided by "IWear" with any of
  the following names:

    IWearConfig.cmake
    iwear-config.cmake

  Add the installation prefix of "IWear" to CMAKE_PREFIX_PATH or set
  "IWear_DIR" to a directory containing one of the above files.  If "IWear"
  provides a separate development package or SDK, be sure it has been
  installed.
~~~


**To Reproduce**

~~~
mamba create -n wearablesdev idyntree librobometry yarp-cxx cmake pkg-config compilers
mamba activate wearablesdev 
git clone https://github.com/robotology/wearables
cd wearables
mkdir build
cd build
cmake ..
cmake .
~~~



**Expected behavior**

The project should not fail at the second configuration.

**Screenshots**

N.a. .

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04, dependencies installed with conda

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ option(ENABLE_HapticGlove "Flag that enables building Haptic Sense Glove wearabl
 
 # Flag to enable IWearFrameVisualizer module
 find_package(iDynTree QUIET)
-option(ENABLE_FrameVisualizer "Flag that enables building IWearFrameVisualizer module" OFF)
+option(ENABLE_FrameVisualizer "Flag that enables building IWearFrameVisualizer module" ${iDynTree_FOUND})
 
 # Flag to enable IWearLogger device
 find_package(robometry QUIET)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,23 +65,6 @@ if(WIN32)
   add_compile_definitions(_SILENCE_EXPERIMENTAL_FILESYSTEM_DEPRECATION_WARNING)
 endif()
 
-add_subdirectory(interfaces)
-add_subdirectory(impl)
-
-# Prepare YARP wrapper and devices
-find_package(YARP 3.2 REQUIRED)
-set(YARP_FORCE_DYNAMIC_PLUGINS ON)
-yarp_configure_plugins_installation(wearable)
-
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
-
-add_subdirectory(msgs)
-add_subdirectory(devices)
-add_subdirectory(wrappers)
-add_subdirectory(app)
-add_subdirectory(modules)
-add_subdirectory(bindings)
-
 # Flag to enable Paexo wearable device
 option(WEARABLES_COMPILE_PYTHON_BINDINGS "Flag that enables building the bindings" OFF)
 
@@ -100,7 +83,7 @@ option(ENABLE_HapticGlove "Flag that enables building Haptic Sense Glove wearabl
 
 # Flag to enable IWearFrameVisualizer module
 find_package(iDynTree QUIET)
-option(ENABLE_FrameVisualizer "Flag that enables building IWearFrameVisualizer module" ${iDynTree_FOUND})
+option(ENABLE_FrameVisualizer "Flag that enables building IWearFrameVisualizer module" OFF)
 
 # Flag to enable IWearLogger device
 find_package(robometry QUIET)
@@ -108,3 +91,22 @@ option(ENABLE_Logger "Flag that enables building Wearable Logger device" ${robom
 
 # Flag to enable ICub wearable device
 option(ENABLE_ICub "Flag that enables building iCub wearable device" ${iDynTree_FOUND})
+
+add_subdirectory(interfaces)
+add_subdirectory(impl)
+
+# Prepare YARP wrapper and devices
+find_package(YARP 3.2 REQUIRED)
+set(YARP_FORCE_DYNAMIC_PLUGINS ON)
+yarp_configure_plugins_installation(wearable)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+add_subdirectory(msgs)
+add_subdirectory(devices)
+add_subdirectory(wrappers)
+add_subdirectory(app)
+add_subdirectory(modules)
+add_subdirectory(bindings)
+
+

--- a/modules/IWearFrameVisualizer/CMakeLists.txt
+++ b/modules/IWearFrameVisualizer/CMakeLists.txt
@@ -18,15 +18,13 @@ endmacro()
 
 set(EXE_TARGET_NAME IWearFrameVisualizerModule)
 
-
-find_package(IWear REQUIRED)
 find_package(YARP REQUIRED)
 find_package(iDynTree REQUIRED)
 
 add_executable(${EXE_TARGET_NAME} src/main.cpp)
 
 target_link_libraries(${EXE_TARGET_NAME} PUBLIC
-    IWear::IWear
+    Wearables::IWear
     YARP::YARP_os
     YARP::YARP_sig
     YARP::YARP_dev

--- a/modules/IWearFrameVisualizer/CMakeLists.txt
+++ b/modules/IWearFrameVisualizer/CMakeLists.txt
@@ -24,7 +24,7 @@ find_package(iDynTree REQUIRED)
 add_executable(${EXE_TARGET_NAME} src/main.cpp)
 
 target_link_libraries(${EXE_TARGET_NAME} PUBLIC
-    Wearables::IWear
+    IWear
     YARP::YARP_os
     YARP::YARP_sig
     YARP::YARP_dev

